x-pack/plugins/alerting/server/alerts_service/lib/set_alerts_to_untracked.ts

```diff
@@ -18,16 +18,34 @@ import {
   ALERT_STATUS_UNTRACKED,
   ALERT_TIME_RANGE,
   ALERT_UUID,
+  AlertStatus,
 } from '@kbn/rule-data-utils';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { AlertingAuthorizationEntity } from '../../authorization/alerting_authorization';
+import type { RulesClientContext } from '../../rules_client';
 
-export interface SetAlertsToUntrackedOpts {
-  indices: string[];
+type EnsureAuthorized = (opts: { ruleTypeId: string; consumer: string }) => Promise<unknown>;
+
+export interface SetAlertsToUntrackedParams {
```

Comment:
It will be nice to not have every attributes as conditional. Maybe we can create a conditional type here to avoid that.

Reply:
Talk to @JiaweiWu, we will keep it like that until we refactor to only use one API like we said [here](https://github.com/elastic/kibana/pull/176264#pullrequestreview-1870960918)